linux/LinuxProcessList.c

```diff
@@ -1843,12 +1868,25 @@ static int scanCPUFreqencyFromSysCPUFreq(LinuxProcessList* this) {
       return -1;
    }
 
-   for (unsigned int i = 0; i < cpus; ++i) {
+   if (configuredCPUs < 0) {
+      /* may be -1 if getting the number of CPUs from sysconf failed */
+      return -1;
+   }
+
+   unsigned int cpus = (unsigned int) configuredCPUs;
+   unsigned int onlineCPUIndex = 0;
+
+   for (unsigned int i = 0; i < cpus && onlineCPUIndex <= this->super.cpuCount - 1; ++i) {
       char pathBuffer[64];
+
+      /* omit CPUs that have been hotplugged off */
+      if (!cpuIsOnline(i))
+         continue;
+
       xSnprintf(pathBuffer, sizeof(pathBuffer), "/sys/devices/system/cpu/cpu%u/cpufreq/scaling_cur_freq", i);
 
       struct timespec start;
-      if (i == 0)
+      if (onlineCPUIndex == 0)
          clock_gettime(CLOCK_MONOTONIC, &start);
 
       FILE* file = fopen(pathBuffer, "r");
```

Comment:
Avoid `fopen` in favour of `openat`; `opendir` the path `/sys/devices/system/cpu` and only scan from there. May combine this with the above and opendir `cpu%d` additionally.